game/songparser.hh

```diff
@@ -4,11 +4,32 @@
 #include "song.hh"
 #include "unicode.hh"
 #include "fs.hh"
-#include <cstdint>
+
 #include <boost/range/adaptor/reversed.hpp>
+
+#include <cstdint>
+#include <regex>
 #include <sstream>
 
+
 namespace SongParserUtil {
+
+#if defined(_MSC_VER)
+	const auto regex_multiline = std::regex_constants::ECMAScript; // MSVC hasn't implemented multiline.
+#else
+	const auto regex_multiline = std::regex::multiline;
+#endif
+
+
+
+	// There is some weird bug with std::regex and boost::locale on libc++ that makes regex fail if a global locale with a collation facet has been installed before instantiating patterns.
+	const static std::regex iniParseLine(R"(^[^\S^\r\n]*([a-zA-Z0-9._-]+)[^\S^\r\n]*=[^\S^\r\n]*([^\n\r]*?)(?=[^\S^\r\n]*$))", regex_multiline);
+	const static std::regex iniCheckHeader(R"(^[^\S^\r\n]*\[song\][^\S^\r\n]*(?:$|[;#]))", regex_multiline);
+	const static std::regex brTag(R"(<br>|<br[ ]*/?>)", std::regex_constants::icase);
+	const static std::regex richTags(
+	  R"(</?(b|i|u|s|size|font|align|gradient|sub|sup|link)(=[^>]*)?( [^>]*)?>|<(color)(=[^>]*)?>|</(color)>)", 
+	  std::regex_constants::icase
+	);
```

Comment:
Let's add some documentation (can be in code) for these huge regexes to define what we wanted to look for.
Everyone can read regex, but not everyone knows what they mean